src/Skia/Avalonia.Skia/FramebufferRenderTarget.cs

```diff
@@ -206,24 +220,73 @@ public PixelFormatConversionShim(SKImageInfo destinationInfo, IntPtr framebuffer
             /// <summary>
             /// Handler to start conversion via surface copy.
             /// </summary>
-            public IDisposable SurfaceCopyHandler { get; }
+            public IDisposable SurfaceCopyHandler { get => Disposable.Create(CopySurface); }
+
+            public SurfaceOrientation Orientation => _orientation;
 
             /// <inheritdoc />
             public void Dispose()
             {
                 Surface.Dispose();
                 _bitmap.Dispose();
             }
-
             /// <summary>
             /// Convert and copy surface to a framebuffer.
             /// </summary>
             private void CopySurface()
             {
                 using (var snapshot = Surface.Snapshot())
                 {
-                    snapshot.ReadPixels(_destinationInfo, _framebufferAddress, _destinationInfo.RowBytes, 0, 0,
-                        SKImageCachingHint.Disallow);
+                    if (Orientation != SurfaceOrientation.Normal)
+                    {
+                        // rotation or flipping required
+                        int width;
+                        int height;
+
+                        if (Orientation == SurfaceOrientation.Rotated180)
+                        {
+                            width = snapshot.Width;
+                            height = snapshot.Height;
+                        }
+                        else
+                        {
+                            width = snapshot.Height;
+                            height = snapshot.Width;
+                        }
```

Comment:
Should the Unknown Orientation swap the width and height?

Reply:
Maybe the Unknown value should not be exposed in the API...
The default orientation is intended to be Normal.
The Unknown value was not intended to be used other than internally; e.g, to force an initial update as used in LibinputBackend.cs:36